packages/webview_flutter/webview_flutter_web/lib/webview_flutter_web.dart

```diff
@@ -0,0 +1,282 @@
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<String?> currentUrl() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<String> evaluateJavascript(String javascript) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<int> getScrollX() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<int> getScrollY() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<String?> getTitle() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> goBack() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> goForward() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> loadUrl(String url, Map<String, String>? headers) async {
+    _element.src = url;
+  }
+
+  @override
+  Future<void> reload() {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> removeJavascriptChannels(Set<String> javascriptChannelNames) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> runJavascript(String javascript) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<String> runJavascriptReturningResult(String javascript) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> scrollBy(int x, int y) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> scrollTo(int x, int y) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> updateSettings(WebSettings setting) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> loadFile(String absoluteFilePath) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> loadHtmlString(
+    String html, {
+    String? baseUrl,
+  }) async {
+    _element.src = 'data:text/html,' + Uri.encodeFull(html);
+  }
+
+  @override
+
+  /// Makes a specific HTTP request ands loads the response in the webview.
+  ///
+  /// [WebViewRequest.method] must be one of the supported HTTP methods
+  /// in [WebViewRequestMethod].
+  ///
+  /// If [WebViewRequest.headers] is not empty, its key-value pairs will be
+  /// added as the headers for the request.
+  ///
+  /// If [WebViewRequest.body] is not null, it will be added as the body
+  /// for the request.
+  ///
+  /// Throws an ArgumentError if [WebViewRequest.uri] has empty scheme.
+  Future<void> loadRequest(WebViewRequest request) async {
+    if (!request.uri.hasScheme) {
+      throw ArgumentError('WebViewRequest#uri is required to have a scheme.');
+    }
+    final HttpRequest httpReq = await _httpRequestFactory.request(request.uri.toString(),
+        method: request.method.serialize(), requestHeaders: request.headers, sendData: request.body);
+    final String contentType = httpReq.getResponseHeader('content-type') ?? 'text/html';
+    print(contentType);
+    _element.src = 'data:$contentType,' + Uri.encodeFull(httpReq.responseText ?? '');
+  }
+}
+
+/// Factory class for creating [HttpRequest] instances.
+class HttpRequestFactory {
```

Comment:
(This is used to be mocked in tests, to ensure that the `WebViewRequest` is correctly parsed; I have a comment about this approach below)

Reply:
The only reason this factory exists, is so that `HttpRequest.request` can be mocked in the tests. 
Ideally we wouldn't have this class at all, but I'm not sure how else I'd mock it. 

Having the user supply the HttpRequest object is not really an option, because as far as I can see, the http request is fired immediately when `HttpRequest.request(...)` is called. 

If you know of a better way to do this, I'd be very interested as it feels like a bit much just for a mock.